core/registry/servicecenter/cache.go

```diff
@@ -235,7 +239,9 @@ func (c *CacheManager) pullMicroserviceInstance() error {
 			lager.Logger.Error("Refresh local instance cache failed", err)
 			continue
 		}
-
+		if len(providerInstances) == 0 {
```

Comment:
necessary? why

Reply:
I guess it will introduce a bug
1. add provider only if there is no service cache when you call Findxxxxx API
2. user just stops all instances, but later they will run again. during that time you fetch providers 
cache and Findxxx from remote,  instance len may become 0,  you delete this provider in cache. 
3. when you call Findxxxxx API again during communication,  there is service key but instance is 0, it will not find from remote again, so you will never add this provider back again in cache. that is the bug